templates/terraform/examples/base_configs/test_file.go.erb

```diff
@@ -120,11 +120,15 @@ func testAccCheck<%= resource_name -%>Destroy(s *terraform.State) error {
 		return err
 	}
 
+	<% if object.custom_code.delete_check -%>
+<%= lines(compile(object.custom_code.delete_check)) -%>
+	<% else -%>
 	_, err = sendRequest(config, "<%= object.read_verb.to_s.upcase -%>", url, nil)
```

Comment:
Should I only replace the contents of the `err=nil { ... }` check? Or do we want the ability to check deletion in other ways than calling the GET endpoint and checking the response?

Reply:
If anything, I'd err the other way and replace more. I think that the loop at the top of the function should stay no matter what, but we could replace everything after that.

That'll keep this similar to defining a custom import, where the user needs to instantiate the config object and write the `return` for the function: https://github.com/GoogleCloudPlatform/magic-modules/blob/master/templates/terraform/custom_import/self_link_as_name.erb